packages/client-sdk-nodejs/src/internal/storage-control-client.ts

```diff
@@ -0,0 +1,159 @@
+    ];
+    this.interceptors = [
+      new HeaderInterceptorProvider(headers).createHeadersInterceptor(),
+      ClientTimeoutInterceptor(StorageControlClient.REQUEST_TIMEOUT_MS),
+    ];
+    this.logger.debug(
+      `Creating storage control client using endpoint: '${props.credentialProvider.getControlEndpoint()}`
+    );
+    this.clientWrapper = new IdleGrpcClientWrapper({
+      clientFactoryFn: () =>
+        new grpcControl.ScsControlClient(
+          props.credentialProvider.getControlEndpoint(),
+          props.credentialProvider.isControlEndpointSecure()
+            ? ChannelCredentials.createSsl()
+            : ChannelCredentials.createInsecure()
+        ),
+      loggerFactory: props.configuration.getLoggerFactory(),
+      maxIdleMillis: props.configuration
+        .getTransportStrategy()
+        .getGrpcConfig()
+        .getDeadlineMillis(),
+    });
+  }
+  close() {
+    this.logger.debug('Closing storage control client');
+    this.clientWrapper.getClient().close();
+  }
+
+  public async createStore(name: string): Promise<CreateStore.Response> {
+    try {
+      validateStoreName(name);
+    } catch (err) {
+      return this.cacheServiceErrorMapper.returnOrThrowError(
+        err as Error,
+        err => new CreateStore.Error(err)
+      );
+    }
+    this.logger.debug(`Creating store: ${name}`);
+    const request = new grpcControl._CreateStoreRequest({
+      store_name: name,
+    });
+    return await new Promise<CreateStore.Response>((resolve, reject) => {
+      this.clientWrapper
+        .getClient()
+        .CreateStore(
+          request,
+          {interceptors: this.interceptors},
+          (err, _resp) => {
+            if (err) {
+              if (err.code === Status.ALREADY_EXISTS) {
+                resolve(new CreateStore.AlreadyExists());
+              } else {
+                this.cacheServiceErrorMapper.resolveOrRejectError({
+                  err: err,
+                  errorResponseFactoryFn: e => new CreateStore.Error(e),
+                  resolveFn: resolve,
+                  rejectFn: reject,
+                });
+              }
+            } else {
+              resolve(new CreateStore.Success());
+            }
+          }
+        );
+    });
+  }
+
+  public async deleteStore(name: string): Promise<DeleteStore.Response> {
+    try {
+      validateStoreName(name);
+    } catch (err) {
+      return this.cacheServiceErrorMapper.returnOrThrowError(
+        err as Error,
+        err => new DeleteStore.Error(err)
+      );
+    }
+    const request = new grpcControl._DeleteStoreRequest({
+      store_name: name,
+    });
+    this.logger.debug(`Deleting store: ${name}`);
+    return await new Promise<DeleteStore.Response>((resolve, reject) => {
+      this.clientWrapper
+        .getClient()
+        .DeleteStore(
+          request,
+          {interceptors: this.interceptors},
+          (err, _resp) => {
+            if (err) {
```

Comment:
I think NotFoundError is supposed to be caught here?

Reply:
The server returns a not found error when the store doesn't exists. 
Added an integration test to verify the same.